x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_color_property.test.tsx

```diff
@@ -578,3 +578,44 @@ test('Should read out ordinal type correctly', async () => {
   expect(ordinalColorStyle2.isOrdinal()).toEqual(true);
   expect(ordinalColorStyle2.isCategorical()).toEqual(false);
 });
+
+describe('renderFieldMetaPopover', () => {
+  const nonGeoJsonField = Object.create(mockField);
```

Comment:
Writing tests with arrays like this unnecessarily couples test cases together and makes reading them much harder. How about this.

```
describe('renderFieldMetaPopover', () => {
  test('Should enable toggle when field is backed by geojson-source', () => {
    const colorStyle = makeProperty(
      {
        color: 'Blues',
        type: undefined,
        fieldMetaOptions,
      },
      undefined,
      mockField
    );

    const legendRow = colorStyle.renderFieldMetaPopover(() => {});
    expect(legendRow).toMatchSnapshot();
  });

  test('Should disable toggle when field is not backed by geojson source', () => {
    const nonGeoJsonField = Object.create(mockField);
    nonGeoJsonField.canReadFromGeoJson = () => {
      return false;
    };
    const colorStyle = makeProperty(
      {
        color: 'Blues',
        type: undefined,
        fieldMetaOptions,
      },
      undefined,
      nonGeoJsonField
    );

    const legendRow = colorStyle.renderFieldMetaPopover(() => {});
    expect(legendRow).toMatchSnapshot();
  });
});
```